Remotion/Core/Core/Reflection/TypeDiscovery/AssemblyLoading/RegexAssemblyLoaderFilter.cs

```diff
@@ -87,9 +87,9 @@ public bool ShouldConsiderAssembly (AssemblyName assemblyName)
       switch (_matchTarget)
       {
         case MatchTargetKind.SimpleName:
-          return _matchExpression.IsMatch (assemblyName.Name);
+          return _matchExpression.IsMatch (assemblyName.Name!);
         default:
-          return _matchExpression.IsMatch (assemblyName.FullName);
+          return _matchExpression.IsMatch (assemblyName.FullName!);
```

Comment:
Right, the `!` is not needed for `.FullName` confused it with `Assembly.FullName`, which is nullable. What should be the content of the TODO? Imo the `assemblyName` and `assemblyName.Name` (conditionally) should have argchecks.

Reply:
if the bang's gone, the todo is gone.

Reply:
So the bang needed on 90 for `.Name` is fine without a check? And the missing argcheck for the public method?

Reply:
Damn. I didn't read the whole sentence.

No bang for FullName.

Bang + Bug-Todo for Name. The bug is that we have to deal with the theoretical option of Name being null or be able to add a sensible check/guard that Name cannot be null in our scenario. E.g. in case this code deals only with persistent assemblies and thus Name cannot be null.